src/main/java/mate/academy/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,78 @@
+package mate.academy.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dao.shoppingcart.AddCartItemRequestDto;
+import mate.academy.dao.shoppingcart.ShoppingCartResponseDto;
+import mate.academy.dao.shoppingcart.UpdateCartItemRequestDto;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.repository.ShoppingCartRepository;
+import mate.academy.service.ShoppingCartService;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping cart management", description = "Endpoints for managing shopping cart")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/cart")
```

Comment:
add `server.servlet.context-path=/api` to application.properties
```suggestion
@RequestMapping("/cart")
```